esmvaltool/diag_scripts/lifetime/lifetime.py

```diff
@@ -0,0 +1,1498 @@
+            axes = fig.add_subplot()
+            plot_kwargs = self._get_plot_kwargs(plot_type, base_dataset)
+            plot_kwargs['axes'] = axes
+            plot_zonalmean = plot_func(cube, **plot_kwargs)
+
+            # Print statistics if desired
+            self._add_stats(plot_type, axes,
+                            self._check_cube_dimensions(cube, plot_type),
+                            dataset)
+
+            # Setup colorbar
+            fontsize = self.plots[plot_type]['fontsize']
+            colorbar = fig.colorbar(plot_zonalmean, ax=axes,
+                                    **self._get_cbar_kwargs(plot_type))
+            colorbar.set_label(self._get_cbar_label(plot_type, dataset),
+                               fontsize=fontsize)
+            colorbar.ax.tick_params(labelsize=fontsize)
+
+            # Customize plot
+            axes.set_title(self._get_label(dataset))
+            fig.suptitle(f"{dataset['long_name']} ({dataset['start_year']}-"
+                         f"{dataset['end_year']})")
+            axes.set_xlabel('latitude [°N]')
+            z_coord = cube.coord(axis='Z')
+            axes.set_ylabel(f'{z_coord.long_name} [{z_coord.units}]')
+            if self.plots[plot_type]['log_y']:
+                axes.set_yscale('log')
+                axes.get_yaxis().set_major_formatter(
+                    FormatStrFormatter('%.1f'))
+            if self.plots[plot_type]['show_y_minor_ticklabels']:
+                axes.get_yaxis().set_minor_formatter(
+                    FormatStrFormatter('%.1f'))
+            else:
+                axes.get_yaxis().set_minor_formatter(NullFormatter())
+            self._process_pyplot_kwargs(plot_type, dataset)
+
+            # Rasterization
+            if self.plots[plot_type]['rasterize']:
+                self._set_rasterized([axes])
+
+        # File paths
+        plot_path = self.get_plot_path(plot_type, base_dataset)
+        netcdf_path = get_diagnostic_filename(Path(plot_path).stem, self.cfg)
+
+        return (plot_path, {netcdf_path: cube})
+
+    def _process_pyplot_kwargs(self, plot_type, dataset):
+        """Process functions for :mod:`matplotlib.pyplot`."""
+        pyplot_kwargs = self.plots[plot_type]['pyplot_kwargs']
+        for (func, arg) in pyplot_kwargs.items():
+            if isinstance(arg, str):
+                arg = self._fill_facet_placeholders(
+                    arg,
+                    dataset,
+                    f"pyplot_kwargs of {plot_type} '{func}: {arg}'",
+                )
+            if arg is None:
+                getattr(plt, func)()
+            else:
+                getattr(plt, func)(arg)
+
+    def _check_cube_dimensions(self, cube, plot_type):
+        """Check that cube has correct dimensional variables."""
+        expected_dimensions_dict = {
+            'annual_cycle': (['month_number'],),
+            'map': (['latitude', 'longitude'],),
+            'zonalmean': (['latitude', self.z_coord]),
```

Comment:
For the monitoring diagnostics you could specify different options here, like `(['latitude', 'air_pressure'], ['latitude', 'altitude'])`, which is probably safer here. Also, there is a comma missing:

```suggestion
            'zonalmean': (['latitude', self.z_coord],),
```

Without the comma this is not interpreted as a tuple